File: wiki_java/src/main/java/com/github/klee0kai/wiki/provide/identifiers/PresentersModule.java
package com.github.klee0kai.wiki.provide.identifiers;

import com.github.klee0kai.stone.annotations.module.Module;

import java.util.concurrent.ThreadPoolExecutor;

@Module
public abstract class PresentersModule {

    @MyQualifier
    public abstract FeaturePresenter provideFeaturePresenter(
            @ThreadQualifier(type = ThreadQualifier.ThreadType.Main) ThreadPoolExecutor executor
    );

    public abstract FeaturePresenter provideFeaturePresenter(
            ScreenId screenId,
            LoginId loginId
    );

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/provide/identifiers/PresentersModule.kt
package com.github.klee0kai.wiki.provide.identifiers

import com.github.klee0kai.stone.annotations.module.Module
import java.util.concurrent.ThreadPoolExecutor

@Module
abstract class PresentersModule {

    @MyQualifier
    abstract fun provideFeaturePresenter(
        @ThreadQualifier(type = ThreadQualifier.ThreadType.Main) executor: ThreadPoolExecutor
    ): FeaturePresenter

    abstract fun provideFeaturePresenter(
        screenId: ScreenId,
        loginId: LoginId
    ): FeaturePresenter

}
